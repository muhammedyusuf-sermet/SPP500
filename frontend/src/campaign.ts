export interface ICampaignState {
	Id?: string,
	Name: string,
	Description: string;
	// Monsters: Monster.IMonsterState[];
}
