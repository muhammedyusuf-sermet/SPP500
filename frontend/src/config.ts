export const API_URL = "https://http://3.18.65.138:3000";