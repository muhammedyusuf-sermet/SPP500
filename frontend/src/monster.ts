export enum Size {
	Tiny = "Tiny",
	Small = "Small",
	Medium = "Medium",
	Large = "Large",
	Huge = "Huge",
	Gargantuan = "Gargantuan"
}

export enum MonsterType {
	Beast = "Beast",
	Dragon = "Dragon",
	Monstrosity = "Monstrosity",
	Humanoid = "Humanoid",
	Fiend = "Fiend",
	Undead = "Undead",
	Giant = "Giant",
	Elemental = "Elemental",
	SwarmOfTinyBeasts = "SwarmOfTinyBeasts",
	Construct = "Construct",
	Celestial = "Celestial",
	Aberration = "Aberration",
	Fey = "Fey",
	Plant = "Plant",
	Ooze = "Ooze"
}

export enum MonsterRace {
	AnyRace = "AnyRace",
	Devil = "Devil",
	Demon = "Demon",
	Human = "Human",
	Shapechanger = "Shapechanger",
	Goblinoid = "Goblinoid",
	Titan = "Titan",
	Gnoll = "Gnoll",
	Gnome = "Gnome",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Orc = "Orc",
	Kobold = "Kobold",
	Lizardfolk = "Lizardfolk",
	Merfolk = "Merfolk",
	Sahuagin = "Sahuagin",
	Grimlock = "Grimlock"
}

export enum Alignment {
	Unaligned = "Unaligned",
	AnyAlignment = "AnyAlignment",
	AnyNonGoodAlignment = "AnyNonGoodAlignment",
	AnyNonEvilAlignment = "AnyNonEvilAlignment",
	AnyNonLawfulAlignment = "AnyNonLawfulAlignment",
	AnyNonChaoticAlignment = "AnyNonChaoticAlignment",
	AnyGoodAlignment = "AnyGoodAlignment",
	AnyEvilAlignment = "AnyEvilAlignment",
	AnyLawfulAlignment = "AnyLawfulAlignment",
	AnyChaoticAlignment = "AnyChaoticAlignment",
	AnyNeutralAlignment = "AnyNeutralAlignment",
	LawfulGood = "LawfulGood",
	LawfulNeutral = "LawfulNeutral",
	LawfulEvil = "LawfulEvil",
	NeutralGood = "NeutralGood",
	Neutral = "Neutral",
	NeutralEvil = "NeutralEvil",
	ChaoticGood = "ChaoticGood",
	ChaoticNeutral = "ChaoticNeutral",
	ChaoticEvil = "ChaoticEvil"
}

export enum Environment {
	Arctic = "Arctic",
	Coastal = "Coastal",
	Desert = "Desert",
	Forest = "Forest",
	Grassland = "Grassland",
	Hill = "Hill",
	Mountain = "Mountain",
	Swamp = "Swamp",
	Underdark = "Underdark",
	Underwater = "Underwater",
	Urban = "Urban"
}

export enum MonsterAction {
	SpecialAbility = "SpecialAbility",
	Action = "Action",
	LegendaryAction = "LegendaryAction"
}

export interface IMonsterAbilities {
	name: string,
	description: string,
}

export enum TypeOfAction {
	Ranged = "Ranged",
	Melee = "Melee",
}

export interface IMonsterActions {
	name: string,
	description: string,
	usageCount: number,
	legendaryAction: boolean,
	typeOfAction?: TypeOfAction,
	toHit: number,
	reach: number,
	targets: number,
	averagedDamage: number,
	damageDice: number,
	damageConstant: number,

}

export interface IMonsterState {
	Id?: number,
	Name: string,
	Size?: Size | string;
	Type?: MonsterType | string;
	Race?: MonsterRace | string;
	Environment?: Environment | string;
	Alignment?: Alignment | string;

	ArmorClass?: number;
	HitPoints?: number;
	HitPointDistribution?: string;

	Speed?: string;
	Languages?: string;

	DamageVulnerabilities?: string;
	DamageResistances?: string;
	DamageImmunities?: string;
	ConditionImmunities?: string;

	ChallengeRating?: number;

	AbilityScores: IMonsterAbilityScoreState;
	Senses: SenseMap;
	Skills: SkillMap;
	SavingThrows: IMonsterSavingThrowState;
	//Actions: IActionState[];
}

export interface SkillMap {
	[skillName: string]: number|undefined;
}

export interface SenseMap {
	[senseName: string]: number|undefined;
}

export interface IMonsterAbilityScoreState {
	[index:string]: number|undefined;
	Strength?: number;
	Dexterity?: number;
	Constitution?: number;
	Intelligence?: number;
	Wisdom?: number;
	Charisma?: number;
}

export interface IMonsterSavingThrowState {
	[index:string]: number|undefined;
	Strength?: number;
	Dexterity?: number;
	Constitution?: number;
	Intelligence?: number;
	Wisdom?: number;
	Charisma?: number;
}

export interface IActionState {
	Name: string,
	Description: string,
	HitBonus?: number,
	Damage?: string,
	DamageBonus?: number,
	Type?: MonsterAction
}
/*export interface IMonster {
	name: string,
	type: MonsterType,
	alignment: MonsterAlignment,
	size: MonsterSize,
	race: MonsterRace,
	environment: MonsterEnvironment,
	resistance?: string,
	damageImmunity?: string,
	conditionImmunity?: string,
	vulnerability?: string,
	armorClass: number,
	hitPoints?: number,
	hitPointDice?: string,
	hitPointDiceAdd?: number,
	speedLand: number,
	speedSwim?: number,
	strStat: number,
	dexStat: number,
	conStat: number,
	intStat: number,
	wisStat: number,
	chaStat: number,
	strSavingThrow?: number,
	dexSavingThrow?: number,
	conSavingThrow?: number,
	intSavingThrow?: number,
	wisSavingThrow?: number,
	chaSavingThrow?: number,
	skillsAthletics: number,
	skillsAcrobatics?: number,
	skillsSleightOfHand?: number,
	skillsStealth?: number,
	skillsArcana?: number,
	skillsHistory?: number,
	skillsInvestigation?: number,
	skillsNature?: number,
	skillsReligion?: number,
	skillsAnimalHandling?: number,
	skillsInsight?: number,
	skillsMedicine?: number,
	skillsPerception?: number,
	skillsSurvival?: number,
	skillsDeception?: number,
	skillsIntimidation?: number,
	skillsPerformance?: number,
	skillsPersuasion?: number,
	sensesBlindsight?: number,
	sensesDarkvision?: number,
	sensesTremorsense?: number,
	sensesTruesight?: number,
	sensesPassivePerception?: number,
	sensesPassiveInvestigation?: number,
	sensesPassiveInsight?: number,
	languages?: string,
	challengeRating: number,
	experiencePoints: number,
	abilities?: IMonsterAbilities[],
	actions?: IMonsterActions[],
}*/