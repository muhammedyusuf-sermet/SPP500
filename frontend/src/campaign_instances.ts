import * as Campaign from "./campaign";

// Dummy array of monsters
import EncounterInstances from './encounter_instances';

let encounterExample: Campaign.ICampaignState;
encounterExample = {
	Id: 0,
	Name: "Campaign",
	Summary: "Bloody hell...",
	Notes: "You gonna die!",
	Encounters: EncounterInstances['EncounterInstances'],
};

let CampaignInstances = new Array();

// Generate a dummy array of encounters with different IDs
for (var i = 0; i < 12; i++) {
	let encounter = {} as Campaign.ICampaignState;
	Object.assign(encounter, encounterExample);
	encounter['Id'] = i;
	CampaignInstances.push(encounter);
}

export default {
    CampaignInstances,
}