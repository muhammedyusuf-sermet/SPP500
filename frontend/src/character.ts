import { ICampaignData } from "./campaign";

export enum CharacterRace {
	Dragonborn = "Dragonborn",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Gnome = "Gnome",
	HalfElf = "HalfElf",
	Halfling = "Halfling",
	HalfOrc = "HalfOrc",
	Human = "Human",
	Tiefling = "Tiefling"
}

export enum CharacterClass {
	Barbarian = "Barbarian",
	Bard = "Bard",
	Cleric = "Cleric",
	Druid = "Druid",
	Fighter = "Fighter",
	Monk = "Monk",
	Paladin = "Paladin",
	Ranger = "Ranger",
	Rogue = "Rogue",
	Sorcerer = "Sorcerer",
	Warlock = "Warlock",
	Wizard = "Wizard"
}

export interface ICharacterData {
	Id?: number;
	Name: string;
	Level?: number;
	Race?: CharacterRace;
	Class?: CharacterClass;
	MaxHealth?: number;
	ArmorClass?: number;
	Notes?: string;

	Campaigns?: ICampaignData[];
}
