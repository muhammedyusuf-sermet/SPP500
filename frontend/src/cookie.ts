import { API_URL } from "./config";
import Store from 'electron-store';

export class CookieManager {
	private constructor(){
	}

	private static store = new Store<string>();

	public static UserToken(cookieToCheck:string):string {
		// undefined if user is not logged in.
		var cookieVal = CookieManager.store.get(cookieToCheck);

		if (cookieVal === undefined)
			return cookieVal

		var request = require("request");
		var options = { method: 'GET',
			url: API_URL + '/verify',
			timeout: 2000,
			headers:
			{
				'Cache-Control': 'no-cache',
				'Content-Type': 'application/json',
				'Authorization': cookieVal
			},
			body: {},
			json: true
		};

		request(options, function (error: string, response: string, body: {status: number, message: string}) {
			if (error) {
				return undefined
			}
			else {
				return (body.status === 201) ? "Valid cookie" : undefined // Success
			}
		});
		return CookieManager.store.get(cookieToCheck);
	}

	public static SetStringCookie(name:string, data:string) {
		CookieManager.store.set(name, data);
	}

	public static RemoveCookie(name:string) {
		CookieManager.store.delete(name);
	}
}
