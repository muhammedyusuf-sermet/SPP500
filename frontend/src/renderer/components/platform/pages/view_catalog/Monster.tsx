import * as React from 'react';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import {MonsterDetails} from './MonsterDetails';
import * as MonsterInterface from '../../../../../monster';

// Dummy array of monsters
import MonsterInstances from '../../../../../monster_instances';

export interface IMonsterState {
	viewMonster: boolean,
	editMonster: boolean,
	selectedMonster: MonsterInterface.IMonster,
}

export class Monster extends React.Component<any, IMonsterState> {
	constructor(props: any) {
		super(props);
		this.state = {
			viewMonster: false,
			editMonster: false,
			selectedMonster: {} as MonsterInterface.IMonster,
		}
		this.resetState = this.resetState.bind(this);
	}

	resetState() {
		this.setState({ selectedMonster: {} as MonsterInterface.IMonster});
		this.setState({ viewMonster: false});
		this.setState({ editMonster: false});
	}

	view = (monster: MonsterInterface.IMonster) => {
		this.setState({ selectedMonster: monster});
		this.setState({ viewMonster: true});
	}

	edit = (monster: MonsterInterface.IMonster) => {
		this.setState({ selectedMonster: monster});
		this.setState({ editMonster: true});
	}

	render() {
		if(!this.state.viewMonster && !this.state.editMonster){
			return (
				<div className= "layout card-grid">
					<Grid container spacing={40}>
						{MonsterInstances['MonsterInstances'].map(monster => (
							<Grid item key={monster.id} sm={6} md={4} lg={3}>
								<Card className="card">
								  <CardMedia
									className="card-media"
									image="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22288%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20288%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_164edaf95ee%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A14pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_164edaf95ee%22%3E%3Crect%20width%3D%22288%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%2296.32500076293945%22%20y%3D%22118.8%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E" // eslint-disable-line max-len
									title="Image title"
								  />
								  <CardContent className="card-content">
									<Typography gutterBottom variant="h5" component="h2">
									  {monster.name} {monster.id}
									</Typography>
									<Typography>
									  This is a {MonsterInterface.MonsterSizeNames.get(monster.size)} {MonsterInterface.MonsterRaceNames.get(monster.race)}, and is a type of {MonsterInterface.MonsterTypeNames.get(monster.type)} monsters.
									</Typography>
								  </CardContent>
								  <CardActions>
									<Button size="small" color="primary" onClick={() => this.view(monster)}>
									  View
									</Button>
									<Button size="small" color="primary" onClick={() => this.edit(monster)}>
									  Edit
									</Button>
								  </CardActions>
								</Card>
							</Grid>
						))}
					</Grid>
				</div>
			)
		}
		else if(this.state.viewMonster){
			return (<MonsterDetails monster={this.state.selectedMonster} resetParentState={this.resetState}/>);
		}
		else{
			// Todo: To enable editing, just use the same component as MonsterDetails:
			// enable edit button, remove readOnly, and add the code to submit the changes
			return (<div>Editing Feature is not yet enabled. Please check again later.</div>);
		}
	}
}
