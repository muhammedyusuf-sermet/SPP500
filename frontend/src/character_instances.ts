import { CharacterRace, CharacterClass, ICharacterData } from "./character";


let characterExample: ICharacterData;
characterExample = {
	Id: 1,
	Name: "Character",
	Level: 3,
	Race: CharacterRace.Gnome,
	Class: CharacterClass.Paladin,
	MaxHealth: 23,
	ArmorClass: 12,
	Notes: 'aggressive',

	Campaigns: []
};

export const CharacterInstances: ICharacterData[] = new Array();

// Generate a dummy array of characters with different IDs
for (var i = 0; i < 12; i++) {
	let character = {} as ICharacterData;
	Object.assign(character, characterExample);
	character.Id = i + 1;
	CharacterInstances.push(character);
}