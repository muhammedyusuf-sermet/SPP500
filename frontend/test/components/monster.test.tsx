import * as React from "react"
import * as nock from 'nock';
import { shallow, ShallowWrapper } from 'enzyme';
import {Monster, IMonsterState} from '../../src/renderer/components/platform/pages/view_catalog/Monster';
import * as MonsterInterface from '../../src/monster';
import {API_URL} from '../../src/config'

jest.mock('../../src/cookie');

describe('Test the Monster View Details', () => {
	let monsterInstance: ShallowWrapper<any, IMonsterState, Monster>;

	beforeEach(() => {
		monsterInstance = shallow(<Monster/>);
	})

	it('renders without crashing', () => {
		expect(monsterInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(monsterInstance).toMatchSnapshot();
	});

	it('should update the state variables when View button is clicked for a monster', () => {
		let selectedMonster: MonsterInterface.IMonsterState;
		selectedMonster = {} as MonsterInterface.IMonsterState;
		monsterInstance.instance().view(selectedMonster);

		expect(monsterInstance.state('viewMonster')).toEqual(true);
		expect(monsterInstance.state('selectedMonster')).toEqual(selectedMonster);
	});

	it('should update the state variables when Edit button is clicked for a monster', () => {
		let selectedMonster: MonsterInterface.IMonsterState;
		selectedMonster = {} as MonsterInterface.IMonsterState;
		monsterInstance.instance().edit(selectedMonster);

		expect(monsterInstance.state('editMonster')).toEqual(true);
		expect(monsterInstance.state('selectedMonster')).toEqual(selectedMonster);
	});

	it('should update the state variables when resetState function is called', () => {
		monsterInstance.instance().resetState();

		expect(monsterInstance.state('viewMonster')).toEqual(false);
		expect(monsterInstance.state('editMonster')).toEqual(false);
		expect(monsterInstance.state('selectedMonster')).toEqual({} as MonsterInterface.IMonsterState);
	});

	it('should make an GET request to retrieve Monsters when getPaginatedMonsters function is called', () => {
		monsterInstance.instance().getPaginatedMonsters(0);

		nock(API_URL)
		.get('/monster/get/0/12')
		.reply(201, { status: 201, message: 'success', total: 1, content: [] as MonsterInterface.IMonsterState[] });
	});
});
