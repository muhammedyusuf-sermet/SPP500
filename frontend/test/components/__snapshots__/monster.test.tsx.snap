// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Monster View Details renders correctly when the page is loaded 1`] = `
<div
  className="layout card-grid"
  id="view-monsters-container"
>
  <Pagination
    getTotalPages={[Function]}
    onPageChange={[Function]}
  />
  <WithStyles(Grid)
    container={true}
    spacing={40}
  />
</div>
`;
